Update README.md, add architecture.md  and a new diagram (#110)
kyma-project/runtime-watcher: docs/api.md

```yaml
spec:
  resourceToWatch:
    group: operator.kyma-project.io
    version: "*"
    resource: kymas
```

### **spec.labelsToWatch**

> **NOTE:** The resources you're watching must have the `operator.kyma-project.io/watched-by` label. The label also informs where to route requests to.

With the **spec.labelsToWatch** attribute, you can filter the specified Group/Version/Kind (GVK) in **spec.resourceToWatch**. For example, if the specified GVK is `secrets`, then it is useful to filter them by a specific label, otherwise, the Runtime Watcher would send an event to the KCP for every Create/Read/Update/Delete (CRUD) event of any Secret in the Kyma cluster.

```yaml
spec:
  labelsToWatch:
    "operator.kyma-project.io/watched-by": "lifecycle-manager"
    "example.label.to.watch": "true"
  resourceToWatch:
    group: ""
    version: "v1"
    resource: secrets
```

### **spec.field**

It uses either the `spec` or `status` value to define if Runtime Watcher sends an event to KCP if the spec of the specified GVK or the status changes.

## KCP configuration

### **spec.serviceInfo**

> Specifies the name and Namespace, as well as the port, the incoming events are routed to.

Reviewer: potentially simpler syntax:

```suggestion
Specifies to which name, Namespace, and port the incoming events are routed.
```
